src/EntryPoints/Specials/SpecialApproverCategories.php

```diff
@@ -3,36 +3,31 @@
 namespace ProfessionalWiki\PageApprovals\EntryPoints\Specials;
 
 use MediaWiki\MediaWikiServices;
-use ProfessionalWiki\PageApprovals\Adapters\DatabaseApproverRepository;
+use ProfessionalWiki\PageApprovals\Application\ApproverRepository;
 use ProfessionalWiki\PageApprovals\Application\UseCases\GetApproversWithCategories;
+use ProfessionalWiki\PageApprovals\PageApprovals;
 use SpecialPage;
-use PermissionsError;
 use LightnCandy\LightnCandy;
 use WebRequest;
 
 class SpecialApproverCategories extends SpecialPage {
 
-	/**
-	 * @var DatabaseApproverRepository
-	 */
-	private $databaseApproverRepository;
+	private ApproverRepository $approverRepository;
 
 	public function __construct() {
-		parent::__construct( 'ApproverCategories' );
-		$db = MediaWikiServices::getInstance()->getDBLoadBalancer()->getConnection( DB_PRIMARY );
-		$this->databaseApproverRepository = new DatabaseApproverRepository( $db );
+		parent::__construct( 'ApproverCategories', restriction: 'manage-approvers' );
+		$this->approverRepository = PageApprovals::getInstance()->getApproverRepository();
```

Comment:
@waximabbax good to construct our services via our factory

Reply:
The `GetApproversWithCategories` below is ideally as well, though it has more issues, such as lack of dependency injection and unclear reason for existing. It looks more like a repository, but that is another issue.